src/librustc_parse/parser/diagnostics.rs

```diff
@@ -1450,6 +1450,11 @@ impl<'a> Parser<'a> {
         consume_close: ConsumeClosingDelim,
     ) {
         let mut brace_depth = 0;
+
+        // Avoid complaining about main in
+        // `src/test/ui/parser/mismatched-braces/missing-close-brace-in-trait.rs`.
+        *self.sess.reached_eof.borrow_mut() = true;
```

Comment:
Are you sure that we actually reach `Eof` in all cases, e.g. the `return` below?

Reply:
changed

Reply:
I had to change it back. Setting `reached_eof` before all of the `return`s in the `loop` would cause the missing `main` error to come up (!?).